backend/src/api/comment/create_comment.rs

```diff
@@ -34,11 +40,51 @@ pub(super) async fn post<D: SqliteDatabase>(
     .await
 }
 
+async fn extract_alias<D: SqliteDatabase>(
+    mut text: String,
+    state: &Arc<AppState<D>>,
+) -> anyhow::Result<String> {
+    let aliases: Vec<String> = text
+        .split(" ")
+        .filter(|word| word.starts_with("!"))
+        .map(|word| word[1..].to_string())
+        .filter(|word| !word.is_empty())
```

Comment:
This can probably be written as one [`filter_map`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map)

Reply:
@fkaa Could you provide an example how you'd implement it? I am looking at the documentation and did not manage to implement this using `filter_map` unfortunately.

Reply:
This is probably fine as is since `filter_map` expects an `Option` to decide if the value should be filtered.
Since none of these functions return an `Option` I can't think of a clean way to write this with `filter_map`.

Reply:
Filtering out duplicates would be worth a consideration though since we are currently checking the database twice if an alias is used twice.
One simple way of doing that would be using the `itertools` crate like this:
```rust
use itertools::Itertools;

// ...

let aliases: Vec<String> = text
    .split(" ")
    .filter(|word| word.starts_with("!"))
    .map(|word| word[1..].to_string())
    .filter(|word| !word.is_empty())
    .unique()
    .collect();
```